mux.go

```diff
@@ -354,11 +355,11 @@ func (mx *Mux) NotFoundHandler() http.HandlerFunc {
 
 // MethodNotAllowedHandler returns the default Mux 405 responder whenever
 // a method cannot be resolved for a route.
-func (mx *Mux) MethodNotAllowedHandler() http.HandlerFunc {
+func (mx *Mux) MethodNotAllowedHandler(allowedMethods ...string) http.HandlerFunc {
```

Comment:
for this exported method, what is the benefit of the `allowedMethods` argument? the for internal method you're using it to pass the request content methodNotAllowed string array of methods that are allowed, but why for the exported method?

Reply:
I did that initially so other parts on the code could eventually use it. Instead, I follow your recommendation and addressed this way https://github.com/go-chi/chi/pull/471/files#diff-b303fad270323289aeb17e2f0930ffa7R428. Let me know what you think